### PR TITLE
Tox and code coverage.

### PR DESCRIPTION
**Pull request by rk4n**
- Added tox with `py27`, `py35` and `flake8` envs
- Code coverage via `pytest-cov`. Coverage will fail only for new or modified files that are poorly covered by tests (< 75%).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,13 @@
 language: python
 
+cache: pip
```
Comment (vladiskuz) [inline .travis.yml]: This line has created many problem in the past. I spent a lot of time to understand why libvirt installation fails on the Travis CI - It cached an old version of libvirt. I don't recommend to use this.

**Comment by vladiskuz** [comment]
I don't think that coverage should fail when it less than 75%. Theoretically we can have some file with two lines of code that don't need a test. I agree - this is good feature, but I think it shouldn't fail

**Comment by rk4n** [comment]
OK, I’ll drop that. We need to investigate that.

**Comment by rk4n** [comment]
It’s a cumulative coverage for files added or edited by you, not an overall project coverage.
For sure we can do “at least 70”. Better to have that as a prompt.

File: .travis.yml (hunk 1 above)
Comment (rk4n) [inline .travis.yml]: OK, I’ll drop that. We need to investigate that.

**Comment by vladiskuz** [comment]
We can delete commits with name "Adding random file which is not covered by tests. Should fail build." and "Removed uncovered file." because one of it added file and another deleted it. These commits add unnecessary information to the history. And one more minor comment - please, don't use dot "." on the end of the first line in a commit message.

**Comment by rk4n** [comment]
@vladiskuz 
> Theoretically we can have some file with two lines of code that don't need a test.
Add those files to ignored in `.coveragerc`.

**Pull request closed without merging**